### PR TITLE
Code rebase

### PR DESCRIPTION
**Pull request by TheMSB**
I am rebasing the code in an attempt to achieve the following:

- [x] Prevent random errors resulting from failed scoping by applying the use of semicolons.
- [x] Separate the responsibilities of each component to make it easier to analyse the code for bugs and create improvements.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -1,130 +1,62 @@
...
-}
-
-// Require each JS file in the main-process dir
-function loadMainFiles() {
-	var files = glob.sync(path.join(__dirname, 'main-process/**/*.js'))
-	files.forEach(file => require(file))
+    const shouldQuit = app.makeSingleInstance(() => {
+        if (mainWindow) {
+            if (mainWindow.isMinimized()) mainWindow.restore();
+            mainWindow.focus()
```
Comment (matjaz) [inline main.js]: `;`

**Comment by matjaz** [comment]
`npm run lint` fails with this changes.

**Comment by TheMSB** [comment]
> npm run lint fails with this changes.

I am aware, I am still working on this.

**Comment by TheMSB** [comment]
code style should be properly implemented right now. (sans some very minor details)

**Comment by TheMSB** [comment]
@matjaz  if you have any comments to add to this PR I'd like to hear them. I've moved some of the plans I had to the next PR as this one was becoming too big.

**Comment by matjaz** [comment]
LGTM 👍

**Pull request merged**